src/router.ts

```diff
@@ -2,23 +2,90 @@ import Vue from 'vue'
 import Router from 'vue-router'
 import SearchAPI from './views/SearchAPI.vue'
 import CreateAPI from './views/CreateAPI.vue'
-import Home from './views/Home.vue'
-
+import Login from './views/auth/LogIn.vue'
+import Signup from './views/auth/SignUp.vue'
+import forgotPassword from './views/auth/ForgotPassword.vue'
+import * as myAuthenticationPlugin from 'authenticationPlugin/App'
 Vue.use(Router)
-
-export default new Router({
+let router = new Router({
   mode: 'history',
   base: process.env.BASE_URL,
   routes: [
+    {
+      path: '/login',
+      name: 'login',
+      component: Login,
+      meta: {
+        requiresGuest: true
+      }
+    },
+    {
+      path: '/signup',
+      name: 'signup',
+      component: Signup,
+      meta: {
+        requiresGuest: true
+      }
+    },
+    {
+      path: '/forgotpassword',
+      name: 'forgotpassword',
+      component: forgotPassword,
+      meta: {
+        requiresGuest: true
+      }
+    },
     {
       path: '/',
       name: 'SearchAPI',
-      component: SearchAPI
+      component: SearchAPI,
+      meta: {
+        requiresAuth: true
+      }
     },
     {
       path: '/api',
       name: 'CreateAPI',
-      component: CreateAPI
+      component: CreateAPI,
+      meta: {
+        requiresAuth: true
+      }
     }
   ]
 })
+// Nav Guards
+router.beforeEach((to, from, next) => {
```

Comment:
This method is very very bad. Please modularize this. tooo many nested conditions.